src/coreclr/jit/earlyprop.cpp

```diff
@@ -624,7 +624,13 @@ GenTree* Compiler::optFindNullCheckToFold(GenTree* tree, LocalNumberToNullCheckT
             return nullptr;
         }
 
-        GenTree* defRHS = defLoc->GetAssignment()->gtGetOp2();
+        GenTreeOp* ssaDefAsg = defLoc->GetAssignment();
```

Comment:
This is not the only place that calls `GetAssignment`.

Did you check them all and made sure they handled this case correctly?